plutus-th/test/simple.plc.golden

```diff
@@ -2,15 +2,6 @@
   (lam
     ds_1
     (all Bool_matchOut_0 (type) (fun Bool_matchOut_0 (fun Bool_matchOut_0 Bool_matchOut_0)))
-    [
-      [
-        [
-          { ds_1 (fun (all a_2 (type) (fun a_2 a_2)) [(con integer) (con 64)]) }
-          (lam thunk_3 (all a_4 (type) (fun a_4 a_4)) (con 64 ! 2))
-        ]
-        (lam thunk_5 (all a_6 (type) (fun a_6 a_6)) (con 64 ! 1))
-      ]
-      (abs a_7 (type) (lam x_8 a_7 x_8))
-    ]
+    [ [ { ds_1 [(con integer) (con 64)] } (con 64 ! 2) ] (con 64 ! 1) ]
```

Comment:
Correct - this version isn't lazy unless it is needed to preserve the semantics. So `if x then 1 else 2` will not be lazy, nor will `case x of { A a1 -> error; B b2 -> error }` (the case lambdas guard the errors), but `if x then error else 1` will be.

Reply:
Er, in particular, the case example will not be lazy in the sense that we will not insert delays in the body, but it won't force the errors that aren't used because they're inside lambdas.

Reply:
Ah, I didn't realize that your optimization applies here as well, because constants are values. Makes sense.